core/src/test/java/ProcessRunner/ProcessIntegrationTest.java

```diff
@@ -0,0 +1,232 @@
+        }
+
+        // 6. Read the first word of the command output
+        String lastReceived;
+        StringBuilder sb = new StringBuilder();
+        while ((lastReceived = receiver.getNextInput()) != null) {
+            if (!lastReceived.equals("\n")) {
+                sb.append(lastReceived);
+            } else {
+                break;
+            }
+        }
+
+        // The first word read should be "test"
+        assert sb.toString().equals("test");
+    }
+
+    @Test
+    void pythonReadTest() throws IOException, InterruptedException {
+        // 1. Create temporary Python file
+        File pyScript = new File("pythonTemp.py");
+        BufferedWriter bf;
+        bf = new BufferedWriter(new FileWriter(pyScript));
+
+        String[] pythonCode = {
+                "import time\n" +
+                        "\n",
+                "while True:\n",
+                "\tprint(\"test\")\n",
+                "\ttime.sleep(0.05)\n",
+        };
+
+        bf.write("");
+        for (String s : pythonCode) {
+            bf.append(s);
+            bf.flush();
+        }
+
+        // 2. Create Python process
+        ProcessRunner process = ProcessFactory.createPythonProcess(pyScript.getAbsolutePath());
+
+        InputReceiver receiver = new InputReceiver();
+        process.addInputObserver(receiver);
+        process.start();
+
+        // 3. Wait for process to start
+        Thread.sleep(500);
+
+        // 3. Read the first word of the command output
+        String lastReceived;
+        StringBuilder sb = new StringBuilder();
+        while (
+                (lastReceived = receiver.getNextInput()) != null &&
+                        !lastReceived.equals(" ") &&
+                        !lastReceived.equals("\n")
+                ) {
+            sb.append(lastReceived);
+        }
+
+        // First word should be "test"
+        assert sb.toString().equals("test");
+    }
+
+    @Test
+    void pythonWriteTest() throws IOException, InterruptedException {
+        // 1. Create temporary Python file
+        File pyScript = new File("pythonTemp.py");
+        BufferedWriter bf;
+        bf = new BufferedWriter(new FileWriter(pyScript));
+
+        String[] pythonCode = {
+                "import time\n" +
+                        "\n" +
+                        "input(\"\")\n",
+                "print(\"test\")\n",
+        };
+
+        bf.write("");
+        for (String s : pythonCode) {
+            bf.append(s);
+            bf.flush();
+        }
+
+        // 2. Create Python process
+        ProcessRunner process = ProcessFactory.createPythonProcess(pyScript.getAbsolutePath());
+
+        InputReceiver receiver = new InputReceiver();
+        process.addInputObserver(receiver);
+        process.start();
+
+        // 3. Wait for process to start
+        Thread.sleep(250);
+
+        // 4. Output "1" to the process
+        process.outputToScript("1");
+
+        // 5. Wait for process to handle the output
```

Comment:
maybe clarify what is to be output? E.g refer to the script in the beginning or something alike 👍